services/opentok-api.js

```diff
@@ -89,130 +89,168 @@ const tokenOptions = (userType) => {
+        { ...defaultSessionOptions, ...options },
+        (err, session) => {
+          if (err) resolve(err);
 
-        activeSession = session;
-        resolve(session);
-      });
-    }
-    catch (err) {
+          // activeSession = session;
+          resolve(session);
+        }
+      );
+    } catch (err) {
       reject(err);
     }
   });
-}
+};
 
 /**
  * Create an OpenTok token
  * @param {String} userType Host, guest, or viewer
  * @returns {String}
  */
-const createToken = userType => OT.generateToken(activeSession.sessionId, tokenOptions(userType));
+const createToken = (userType, sessionId) =>
+  OT.generateToken(sessionId, tokenOptions(userType));
 
 /**
  * Creates an OpenTok session and generates an associated token
  * @returns {Promise} <Resolve => {Object}, Reject => {Error}>
  */
-const getCredentials = async (userType) => {
+const getCredentials = async (userType, sessionId = null) => {
   return new Promise(async (resolve, reject) => {
-    if (!activeSession) {
+    let session;
+    if (!session) {
       try {
-        await createSession();
-      }
-      catch (err) {
+        session = await createSession();
+      } catch (err) {
         reject(err);
       }
     }
-    const token = createToken(userType);
-    resolve({ apiKey, sessionId: activeSession.sessionId, token });
+    // }
+    const token = createToken(userType, session.sessionId);
+    resolve({ apiKey, sessionId: session.sessionId, token });
   });
-}
+};
 
 /**
  * Start the broadcast and keep the active broadcast in memory
  * @param {Number} streams - The current number of published streams
  * @param {String} [rmtp] - The (optional) RTMP stream url
  * @returns {Promise} <Resolve => {Object} Broadcast data, Reject => {Error}>
  */
-const startBroadcast = async (streams, rmtp) => {
+const startBroadcast = async (
+  streams,
+  rtmp,
+  lowLatency,
+  fhd = false,
+  dvr = false,
+  sessionId
+) => {
   return new Promise((resolve, reject) => {
+    console.log('StartBroadcast API');
 
-    let layout;
-    if (streams > 3) {
-      layout = {
-        type: 'bestFit'
-      };
-    }
-    else {
-      layout = {
-        type: 'custom',
-        stylesheet: customStyle,
-      };
+    const layout = {
+      type: 'bestFit',
+      screenshareType: 'pip',
+    };
+    let dvrConfig = dvr;
+    let lowLatencyConfig = lowLatency;
+    // if (streams > 3) {
+    //   layout = {
+    //     type: 'bestFit',
+    //   };
+    // } else {
+    //   layout = {
+    //     type: 'custom',
+    //     stylesheet: customStyle,
+    //   };
+    // }
+    if (dvrConfig) {
+      lowLatencyConfig = false; // DVR and LL are not compatible
     }
-    const outputs = {
-      hls: {},
+    let outputs = {
+      hls: {
+        dvr: dvrConfig,
+        lowLatency: lowLatencyConfig,
+      },
     };
-    const sessionId = activeSession.sessionId;
 
-    const { serverUrl, streamName } = rmtp;
+    const { serverUrl, streamName } = rtmp;
```

Comment:
typo?

Reply:
it's  actually rtmp, there was a typo in the previous version

Reply:
sorry yes my eyes